scripts/zones/Kazham/npcs/Magriffon.lua

```diff
@@ -1,25 +1,49 @@
 -----------------------------------
 -- Area: Kazham
 -- NPC: Magriffon
--- Standard Info NPC
+-- Involved in Quest: Gullible's Travels, Even More Gullible's Travels,
+-- Location: (I-7)
 -----------------------------------
 
 package.loaded["scripts/zones/Kazham/TextIDs"] = nil;
+require("scripts/globals/settings");
+require("scripts/globals/quests");
+require("scripts/globals/titles");
 require("scripts/zones/Kazham/TextIDs");
 
+
 -----------------------------------
 -- onTrade Action
 -----------------------------------
 
 function onTrade(player,npc,trade)
-end; 
+    if(player:getQuestStatus(OUTLANDS, GULLIBLES_TRAVELS) == QUEST_ACCEPTED) then
+        if(trade:getGil() >= player:getVar("MAGRIFFON_GIL_REQUEST")) then
+            player:startEvent(0x0092);

```

Comment:
Maybe tradeComplete() here? I think we've had problems in the past trying to finish a trade across functions/events